### PR TITLE
lightning: output clearer output message for max-error

### LINKED ISSUE #29813: lightning should notify user if there is error during import
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
Use lightning to import data with errors, and specify max-error parameters >= the number of actually errors.

For example, set max-error = 2 and sql-mode = "STRICT_TRANS_TABLES" for below data with 2 errors:
```
[root@centos76_vm max_error]# cat max_error.t2-schema.sql
/*!40101 SET NAMES binary*/;
CREATE TABLE `t2` (
  `a` int(11) NOT NULL,
  `b` varchar(10) DEFAULT NULL,
  `c` datetime DEFAULT NULL,
  PRIMARY KEY (`a`) /*T![clustered_index] CLUSTERED */,
  UNIQUE KEY `uk_b` (`b`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;
[root@centos76_vm max_error]# cat max_error.t2.000000000.csv
"a","b","c"
1,"b1","2021-11-14 00:00:00"
"xxx","b2","2021-11-15 00:00:00"
3,"b3","0"
4,"b4","2021-11-17 00:00:00"

```

### 2. What did you expect to see? (Required)
There should be some warning in console output and end of the lightning log file to let user know that there are errors during import.

### 3. What did you see instead (Required)
No warning in console output or log file, user might think that the import were successful.
 
max_error

### 4. What is your TiDB version? (Required)
[root@centos76_vm max_error]# /root/tidb/bin/tidb-lightning -V
Release Version: v5.3.0-alpha-1338-g1f55e22-dirty
Git Commit Hash: 1f55e22358a81af755f868835efc6bc5cbd1095c
Git Branch: release-5.3
Go Version: go1.16.6
UTC Build Time: 2021-11-15 07:27:47
Race Enabled: false

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?

Issue Number: close #29813

Problem Summary:

### What is changed and how it works?
- Output an message both in stdout and log if there is any collected errors.
Stdout output:
f2RgbThCBi

ouput in log (at the end of lightning log):
```
[2021/12/21 15:52:52.480 +08:00] [INFO] [restore.go:470] ["the whole procedure completed"] [takeTime=2.087220285s] []
[2021/12/21 15:52:52.480 +08:00] [WARN] [errormanager.go:377] ["Detect 3 data type errors in total, please refer to table `lightning_task_info`.`type_error_v1` for more details"]
```

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/errormanager/errormanager.go [hunk 1]
```diff
@@ -177,6 +182,11 @@ func (em *ErrorManager) RecordTypeError(
 ) error {
 	// elide the encode error if needed.
 	if em.remainingError.Type.Dec() < 0 {
+		threshold := em.configError.Type.Load()
+		if threshold > 0 {
+			encodeErr = errors.Annotatef(encodeErr, "meet errors exceed the max-error threshold %d",
+				em.configError.Type.Load())
```
Comment (kennytm) [inline br/pkg/lightning/errormanager/errormanager.go:188]:
```suggestion
				threshold)
```

**Review by kennytm — APPROVED** [review]
rest lgtm

**Comment by gozssky** [comment]
@glorv  Should we output conflict error count in this PR?

**Comment by glorv** [comment]
> @glorv Should we output conflict error count in this PR?

This depends on whether we collect duplicate error count. The output message only print error types that count > 0, so duplicate error won't output because the value is always now.

**Comment by gozssky** [comment]
> > @glorv Should we output conflict error count in this PR?
> 
> This depends on whether we collect duplicate error count. The output message only print error types that count > 0, so duplicate error won't output because the value is always now.

@glorv I mean should we also collect duplicate error count in `RecordIndexConflictError` and `RecordDataConflictError`.

**Comment by sunzhaoyang** [comment]
output LGTM

**Review by gozssky — APPROVED** [review]

**Pull request merged**